File: Beginner/uniformStrings/uniformStrings.java
//run as java -ea uniformStrings
import java.util.Scanner;

public class uniformStrings{
	public static void main(String[] args){
		int T,count;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=256) : "T must be 1 <= T <= 256";

		while(T>0){
			String s;
			
			s = scan.next();
			assert(s.length()==8) : "Length of s must be 8";

			count = 0;

			for(int i=0;i<7;i++){
				if(s.charAt(i)!=s.charAt(i+1)){
					count++;
				}
			}

			if(count<=2){
				System.out.println("uniform");
			}else{
				System.out.println("non-uniform");
			}
			
			T--;
		}
	}
}

File: Beginner/uniformStrings/uniformStrings.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T,count;

	scanf("%d",&T);
	assert(T>=1 && T<=256);

	while(T>0){
		char s[9];
		scanf("%s",s);

		count = 0;

		for(int i=0;i<7;i++){
			if(s[i]!=s[i+1]){
				count++;
			}
		}

		if(count<=2){
			printf("uniform\n");
		}else{
			printf("non-uniform\n");
		}

		T--;
	}
	
	return 0;
}